### PR TITLE
[IMP] project_timesheet_holidays: improve demo data timesheet

### PR DESCRIPTION
**Pull request by ppr-odoo**
In this commit before, timesheets are not generated for approved time off. So we have entered the timesheet regarding the approved time off.

task-2960247

### CODE REVIEW

**Review by MissingNoShiny — COMMENTED** [review]
Hello, thanks for your work. :slightly_smiling_face: 
Instead of generating the data manually, I'm wondering if we could not just call the method generating the timesheets on the leaves?

**Comment by ppr-odoo** [comment]
> Hello, thanks for your work. 🙂 Instead of generating the data manually, I'm wondering if we could not just call the method generating the timesheets on the leaves?


     <function model="hr.leave" name="_remove_resource_leave">
            <value eval="[ref('hr_holidays.hr_holidays_sl')]"/>
    </function>

     <function model="hr.leave" name="_validate_leave_request">
            <value eval="[ref('hr_holidays.hr_holidays_sl')]"/>
     </function>

I agree with you but we have to remove leave from the resource.calendar.leaves then a timesheet will be generated.
Because resource.calendar.leaves contains a leave record, the [list_work_time_per_day ](https://github.com/odoo/odoo/blob/16.0/addons/project_timesheet_holidays/models/hr_holidays.py#L88 ) method returns empty. it's ok for you? 

FYI  `When the timesheet is created, the resource.calendar.leaves record is  again created`

**Comment by MissingNoShiny** [comment]
Hello, we can remove the resource.calendar.leaves and let them get created again, it seems cleaner to me than generating the data manually :slightly_smiling_face:

**Comment by ppr-odoo** [comment]
@MissingNoShiny i have done the changes.thanks

File: addons/project_timesheet_holidays/data/holiday_timesheets_demo.xml [hunk 1]
```diff
@@ -1,97 +1,14 @@
...
-            <field name="unit_amount">8.00</field>
-            <field name="project_id" model="project.project" eval="obj().env.ref('hr_holidays.hr_holidays_sl_qdp').holiday_status_id.timesheet_project_id.id"/>
-            <field name="task_id" model="project.task" eval="obj().env.ref('hr_holidays.hr_holidays_sl_qdp').holiday_status_id.timesheet_task_id.id"/>
-            <field name="holiday_id" ref="hr_holidays.hr_holidays_sl_qdp"/>
-        </record>
-
+    <data>
+        <function model="hr.leave" name="_remove_resource_leave">
+            <value eval="[ref('hr_holidays.hr_holidays_sl'), ref('hr_holidays.hr_holidays_cl_qdp'), ref('hr_holidays.hr_holidays_sl_qdp')]"/>
+        </function>
```
Comment (xavierbol) [inline addons/project_timesheet_holidays/data/holiday_timesheets_demo.xml:6]: Why do you need to remove the `resouce.calendar.leave` for those leaves and after `_timesheet_create_lines` you recreate those ones? 🤔

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left one comment. 🙂

File: addons/project_timesheet_holidays/data/holiday_timesheets_demo.xml (hunk 1 above)
Comment (xavierbol) [reply addons/project_timesheet_holidays/data/holiday_timesheets_demo.xml:6]: Oh okay, I saw your comment above for Vincent. 🙂

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

Can you squash your commits? Thanks in advance! 🙂

**Comment by ppr-odoo** [comment]
> LGTM, thanks for your work. 🙂
> 
> Can you squash your commits? Thanks in advance! 🙂

Done.Thank you 👍

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+

**Pull request closed without merging**